### PR TITLE
CAT-1746 Implemented an option to change variable names

### PR DESCRIPTION
**Pull request by Manu91**

### CODE REVIEW

**Comment by Manu91** [comment]
ok to test

**Comment by thmq** [comment]
ok to test

**Comment by Bioxar** [comment]
ok to test

**Comment by thmq** [comment]
there is one checkstyle error, please fix this.

**Comment by thmq** [comment]
retest this please

**Comment by Manu91** [comment]
ok to test

File: catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+		nameEditText = (EditText) dialogView.findViewById(R.id.dialog_formula_rename_variable_name_edit_text);
+
+		switch (type) {
+			case USER_LIST:
+				nameEditText.setText(userList.getName());
+				break;
+			case USER_VARIABLE:
+				nameEditText.setText(userVariable.getName());
+				break;
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java]: add a default case

File: catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	private void handleOkButton() {
+		String name = nameEditText.getText().toString();
+
+		switch (type) {
+			case USER_LIST:
+				renameUserList(name);
+				break;
+			case USER_VARIABLE:
+				renameUserVariable(name);
+				break;
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java]: add a default case

File: catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+		if (name.length() == 0) {
+			positiveButton.setEnabled(false);
+		}
+	}
+
+	private boolean isVariableNameValid(String name) {
+		DataContainer currentData = ProjectManager.getInstance().getCurrentProject().getDataContainer();
+
+		if (ProjectManager.getInstance().getCurrentProject().getDataContainer().existProjectVariable(userVariable)) {
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java]: Why don't use currentData?

File: catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+			List<Sprite> sprites = ProjectManager.getInstance().getCurrentProject().getSpriteList();
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/dialogs/RenameVariableDialog.java]: Why don't use currentData?

**Comment by Manu91** [comment]
fixed

**Comment by Manu91** [comment]
ok to test

**Pull request merged**